### PR TITLE
feat: Add tags checking rule - allows specify rules for any tag and validate that

### PR DESCRIPTION
**Pull request by thedaviddias**
Initial PR done by @SqrTT #184

**Short description of what this resolves:**

Add tags checking rule

### CODE REVIEW

File: test/rules/tags-check.spec.js [hunk 1]
```diff
@@ -1,70 +1,63 @@
-/**
- * Copyright (c) 2015, Yanis Wang <yanis.wang@gmail.com>
- * MIT Licensed
- */
 
-var expect  = require("expect.js");
+const expect  = require("expect.js");
```
Comment (Shinigami92) [inline test/rules/tags-check.spec.js:2]: in this file singlequotes seems to be used
we could format the whole codebase with prettier later, so this wont be an issue for now

File: src/rules/tags-check.js [hunk 2]
```diff
@@ -39,80 +39,80 @@ function assign(target) {
...
-            assign(tagsTypings, options);
-        }
+      if (typeof options !== 'boolean') {
+          assign(tagsTypings, options);
+      }
 
-        parser.addListener('tagstart', function (event) {
-            var attrs = event.attrs;
-            var col = event.col + event.tagName.length + 1;
+      parser.addListener('tagstart', function (event) {
```
Comment (Shinigami92) [inline src/rules/tags-check.js:52]: we should convert all of this functions to arrow-functions
but we should do this for the whole codebase and use eslint to find all of this occurrences

Comment (thedaviddias) [reply src/rules/tags-check.js:52]: That's the plan right after we merge all old PRs :)

File: src/rules/tags-check.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ var tagsTypings = {
     }
 };
 
-function assign(target) {
+var assign = function(target) {
```
Comment (Shinigami92) [inline src/rules/tags-check.js:30]: image

**Pull request merged**